rest/api_test.go

```diff
@@ -686,45 +686,59 @@ func TestFunkyDocAndAttachmentIDs(t *testing.T) {
+			origin:       "http://staging.example.com",
+			headerOutput: "http://staging.example.com",
+		},
 
-	reqHeaders := map[string]string{
-		"Origin": "http://example.com",
+		{
+			origin:       "http://hack0r.com",
+			headerOutput: "*",
+		},
 	}
-	response := rt.SendRequestWithHeaders("GET", "/db/", "", reqHeaders)
-	goassert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "http://example.com")
+	for _, tc := range tests {
+		t.Run(tc.origin, func(t *testing.T) {
 
-	// now test a non-listed origin
-	// b/c * is in config we get *
-	reqHeaders = map[string]string{
-		"Origin": "http://hack0r.com",
-	}
-	response = rt.SendRequestWithHeaders("GET", "/db/", "", reqHeaders)
-	goassert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "*")
+			reqHeaders := map[string]string{
+				"Origin": tc.origin,
+			}
 
-	// now test another origin in config
-	reqHeaders = map[string]string{
-		"Origin": "http://staging.example.com",
-	}
-	response = rt.SendRequestWithHeaders("GET", "/db/", "", reqHeaders)
-	goassert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "http://staging.example.com")
+			response := rt.SendRequestWithHeaders("GET", "/db/", "", reqHeaders)
+			assert.Equal(t, tc.headerOutput, response.Header().Get("Access-Control-Allow-Origin"))
+			assertStatus(t, response, http.StatusUnauthorized)
+			require.Contains(t, response.Body.String(), ErrLoginRequired.Message)
 
-	// test no header on _admin apis
-	reqHeaders = map[string]string{
-		"Origin": "http://example.com",
-	}
-	response = rt.SendAdminRequestWithHeaders("GET", "/db/_all_docs", "", reqHeaders)
-	goassert.Equals(t, response.Header().Get("Access-Control-Allow-Origin"), "")
+			response = rt.SendRequestWithHeaders("GET", "/notadb/", "", reqHeaders)
+			assert.Equal(t, tc.headerOutput, response.Header().Get("Access-Control-Allow-Origin"))
+			assertStatus(t, response, http.StatusUnauthorized)
+			require.Contains(t, response.Body.String(), ErrLoginRequired.Message)
 
-	// test with a config without * should reject non-matches
-	sc := rt.ServerContext()
-	sc.config.API.CORS.Origin = []string{"http://example.com", "http://staging.example.com"}
-	// now test a non-listed origin
-	// b/c * is in config we get *
-	reqHeaders = map[string]string{
-		"Origin": "http://hack0r.com",
+			// admin port doesn't have CORS
+			response = rt.SendAdminRequestWithHeaders("GET", "/db/_all_docs", "", reqHeaders)
+			assert.Equal(t, "", response.Header().Get("Access-Control-Allow-Origin"))
+			assertStatus(t, response, http.StatusOK)
+
+			// test with a config without * should reject non-matches
+			sc := rt.ServerContext()
+			defer func() {
+				sc.config.API.CORS.Origin = defaultTestingCORSOrigin
+			}()
+
+			sc.config.API.CORS.Origin = []string{"http://example.com", "http://staging.example.com"}
+			if !base.StringSliceContains(sc.config.API.CORS.Origin, tc.origin) {
+				response = rt.SendRequestWithHeaders("GET", "/{{.keyspace}}/", "", reqHeaders)
```

Comment:
I didn't think we should be using {{.keyspace}} in 3.0.6.  (is this path of the test covered?)

Reply:
We don't assert on the status of the request so CORS still populates the headers.

Anyway I changed it for clarity.